### PR TITLE
Converting more of the x86 HWIntrinsic tests to be generated from a template

### PR DESCRIPTION
**Pull request by tannergooding**
FYI. @CarolEidt, @fiigii

### CODE REVIEW

**Comment by tannergooding** [comment]
The following Sse tests still need to be converted:
```
ConvertToInt32
ConvertToInt64
ConvertToInt32WithTruncation
ConvertToInt64WithTruncation
ConvertToSingle
ConvertToVector128SingleScalar
LoadAlignedVector128
LoadHigh
LoadLow
LoadScalar
LoadVector128
MoveHighToLow
MoveLowToHigh
MoveMask
MoveScalar
Reciprocal
ReciprocalScalar
ReciprocalSqrt
ReciprocalSqrtScalar
SetAllVector128
SetScalar
SetVector128
SetZeroVector
Shuffle
Sqrt
SqrtScalar
StaticCast
Store
StoreAligned
StoreAlignedNonTemporal
StoreHigh
StoreLow
StoreScalar
UnpackHigh
UnpackLow
```

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
Fixed a bug where `CompareEqualOrderedScalar`, `CompareEqualUnorderedScalar`, `CompareNotEqualOrderedScalar`, and `CompareNotEqualUnorderedScalar` could be assigned a non-byteable temporary register, which would then fail the encoding check,

**Comment by tannergooding** [comment]
Rebased to resolve a merge conflict.

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
Ubuntu failures are unrelated: #16134

**Comment by tannergooding** [comment]
This should be ready for review.

**Comment by fiigii** [comment]
Thanks for the work, I will take a look by end of today.

File: tests/src/JIT/HardwareIntrinsics/X86/Shared/BooleanCmpOpTest.template [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            var right = Unsafe.Read<{3}<{2}>>(_dataTable.inArray2Ptr);
+            var result = {0}.{1}(left, right);
+
+            ValidateResult(left, right, result);
+        }}
+
+        public void RunLclFldScenario()
+        {{
+            var test = new BooleanComparisonOpTest__{1}{2}();
+            var result = {0}.{1}(test._fld1, test._fld2);
```
Comment (fiigii) [inline tests/src/JIT/HardwareIntrinsics/X86/Shared/BooleanCmpOpTest.template]: Hmm.. Looks like we need another template for Avx comparisons that take two vectors and an imm value.

Comment (tannergooding) [reply tests/src/JIT/HardwareIntrinsics/X86/Shared/BooleanCmpOpTest.template]: Yes, I think we will end up having at least 5-10 templates to support all tests, but this is much more manageable than 500-1000 individual tests that need handling 😉 

I think it is even fine if the templates are "nearly" identical in some cases, as they are still few in number and it makes the overall logic to maintain them simpler.

**Pull request merged**